app/code/Magento/CustomerGraphQl/Model/Customer/ExtractCustomerData.php

```diff
@@ -101,7 +101,8 @@ public function execute(CustomerInterface $customer): array
             }
         }
         $customerData = array_merge($customerData, $customAttributes);
-
+        //Field is deprecated and should not be exposed on storefront.
+        $customerData['group_id'] = null;
```

Comment:
I see, nice to know the correct file to change in cases like this.
Thank you, @lenaorobei.